packages/advanced-logic/src/extensions/payment-network/bitcoin/mainnet-address-based.ts

```diff
@@ -13,22 +11,11 @@ const BITCOIN_NETWORK = 'mainnet';
  * Important: the addresses must be exclusive to the request
  */
 export default class BitcoinAddressBasedPaymentNetwork extends AddressBasedPaymentNetwork {
-  public constructor() {
-    super(
-      ExtensionTypes.ID.PAYMENT_NETWORK_BITCOIN_ADDRESS_BASED,
-      CURRENT_VERSION,
-      [BITCOIN_NETWORK],
-      RequestLogicTypes.CURRENCY.BTC,
-    );
-  }
-
-  /**
-   * Check if a bitcoin address is valid
-   *
-   * @param address address to check
-   * @returns true if address is valid
-   */
-  protected isValidAddress(address: string): boolean {
-    return walletAddressValidator.validate(address, 'bitcoin', 'prod');
+  public constructor(
+    public extensionId: ExtensionTypes.ID = ExtensionTypes.ID.PAYMENT_NETWORK_BITCOIN_ADDRESS_BASED,
+    public currentVersion: string = CURRENT_VERSION,
+    public supportedNetworks: string[] = [BITCOIN_NETWORK],
```

Comment:
Why do we introduce parameters? Are those parameters injectable with other kinds of values?